### PR TITLE
Added basic spinner tests and modified spinner method docstring

### PR DESCRIPTION
**Pull request by OmriGM**
### Background

- Improve the quality of AutoGPT by adding unit tests
- Started adding unit tests to the Spinner class
### Changes

- Added 5 basic unit tests to the Spinner class
- Modified the docstring of `update_message` method
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: tests/unit/test_spinner.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+- __exit__(self, exc_type, exc_value, exc_traceback) -> None: Stops the spinner animation when used as a context manager.
+- update_message(self, new_message, delay=0.1): Updates the message displayed by the spinner animation.
+
+Fields:
+- spinner: An itertools.cycle object that contains the characters used for the spinner animation.
+- delay: The delay between each spinner update.
+- message: The message to display.
+- running: A boolean value that indicates whether the spinner animation is running.
+- spinner_thread: A threading.Thread object that runs the spin method in a separate thread.
+"""
```
Comment (ntindle) [inline tests/unit/test_spinner.py:26]: Did you do this by hand or with a tool? If a tool, what tool?

**Review by ntindle — APPROVED** [review]

File: tests/unit/test_spinner.py (hunk 1 above)
Comment (OmriGM) [reply tests/unit/test_spinner.py:26]: I used [CodiumAI's VSCode Extension](https://marketplace.visualstudio.com/items?itemName=Codium.codium) which came pretty handy here, and it also helped writing the unit tests!

Comment (ntindle) [reply tests/unit/test_spinner.py:26]: I feel like we are about to become very familiar with that tool

**Comment by ntindle** [comment]
Will you update from base so I can test the code coverage tool before we merge?

**Review by Torantulino — APPROVED** [review]

**Comment by ntindle** [comment]
@OmriGM would love to see some more of these come through if you have the capacity

**Pull request merged**